src/TextArea/TextArea.less

```diff
@@ -0,0 +1,98 @@
+@import (reference) "../less/colors";
+@import (reference) "../less/border";
+@import (reference) "../less/sizing";
+@import (reference) "../less/type-size";
+@import (reference) "../less/type-utilities";
+@import (reference) "../less/spacing";
+
+.TextArea {
+  .padding--left--s;
+  .padding--top--3xs;
+  .text--regular;
+  .border--s(@neutral_silver);
+  position: relative;
+  color: @neutral_black;
+  background: @neutral_white;
+  box-sizing: border-box;
+  width: 100%;
+
+  .TextArea--infoRow {
+    .text--tiny;
+    .text--caps;
+    position: absolute;
+    top: @size_xs;
+    width: 100%;
+    color: @neutral_dark_gray;
+
+    .TextArea--label {
```

Comment:
since this is a label for the TextArea  - and it will only ever have one label - it's fine to have it as `TextArea--label`.
(if there was an InfoRow component that abstracted this stuff away, then it would have to be `TextArea--InfoRow--label`)

but you should move these nested rules out into the top-level scope as Carl pointed out above.